addons/parley/views/parley_graph_edge.gd

```diff
@@ -0,0 +1,63 @@
+# ParleyGraphEdge.gd
+# Represents a connection between two GraphNodes in a GraphEdit
+
+extends Object
+class_name ParleyGraphEdge
+var from_node: ParleyGraphNode
+var from_slot: int
+var from_port: int
+var to_node: ParleyGraphNode
+var to_slot: int
+var to_port: int
+var to_node_name: String
+var from_node_name: String
+var edge_ast: ParleyEdgeAst
+
+var previousFromColor: Color
+var previousToColor: Color
+var selected: bool
+
+func _init(_edge_ast: ParleyEdgeAst, _from_node: ParleyGraphNode, _from_port: int, _to_node: ParleyGraphNode, _to_port: int) -> void:
+	from_node = _from_node
+	from_port = _from_port
+	to_node = _to_node
+	to_port = _to_port
+	from_slot = from_node.get_output_port_slot(from_port)
+	to_slot = to_node.get_input_port_slot(to_port)
+	from_node_name = from_node.name
+	to_node_name = to_node.name
+	edge_ast = _edge_ast
+
+func disconnect_node(graph_view: ParleyGraphView) -> void:
+	if not graph_view:
+		return
+	graph_view.ast.remove_edge(from_node.id, from_port, to_node.id, to_port)
+	graph_view.disconnect_node(from_node_name, from_port, to_node_name, to_port)
+
+func connect_node(graph_view: ParleyGraphView) -> int:
+	if not graph_view:
+		return FAILED
```

Comment:
I've been searching for something like this for ages (I've previously used the specific: `ERR_` codes) - TIL!